examples/AdvancedOperations/AddDynamicSearchAds.php

```diff
@@ -0,0 +1,351 @@
+            'campaign' => $campaignResourceName,
+            'status' => AdGroupStatus::PAUSED,
+            'type' => AdGroupType::SEARCH_DYNAMIC_ADS,
+            'tracking_url_template' => new StringValue(
+                ['value' => 'http://tracker.examples.com/traveltracker/{escapedlpurl}']
+            ),
+            'cpc_bid_micros' => new Int64Value(['value' => 10000000])
+        ]);
+
+        // Creates an ad group operation.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setCreate($adGroup);
+
+        // Issues a mutate request to add the ad groups.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        /** @var MutateAdGroupsResponse $adGroupResponse */
+        $adGroupResponse = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        $adGroupResourceName = $adGroupResponse->getResults()[0]->getResourceName();
+        printf("Added ad group named '%s'.%s", $adGroupResourceName, PHP_EOL);
+
+        return $adGroupResourceName;
+    }
+
+    /**
+     * Creates an expanded dynamic search ad.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     * @param string $adGroupResourceName the ad group resource name
+     */
+    private static function createExpandedDSA(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adGroupResourceName
+    ) {
+        $adGroupAd = new AdGroupAd([
+            'ad_group' => $adGroupResourceName,
+            'status' => AdGroupAdStatus::PAUSED,
+            'ad' => new Ad([
+                'expanded_dynamic_search_ad' => new ExpandedDynamicSearchAdInfo([
+                    'description' => new StringValue(['value' => 'Buy tickets now!'])
+                ])
+            ])
+        ]);
+
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+
+        // Issues a mutate request to add the ad group ads.
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+        /** @var MutateAdGroupAdsResponse $adGroupAdResponse */
+        $adGroupAdResponse = $adGroupAdServiceClient->mutateAdGroupAds(
+            $customerId,
+            [$adGroupAdOperation]
+        );
+
+        $adGroupAdResourceName = $adGroupAdResponse->getResults()[0]->getResourceName();
+        printf("Added ad group ad named '%s'.%s", $adGroupAdResourceName, PHP_EOL);
+
+        return $adGroupAdResourceName;
+    }
+
+    /**
+     * Creates webpage targeting criteria for the DSA.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     * @param string $adGroupResourceName the resource name of the ad group
+     */
+    private static function createWebPageCriteria(
```

Comment:
"criteria" is plural of "criterion". We only create one criterion here so I would change to criterion.

Reply:
Done.